vuvuzela_test.go

```diff
@@ -0,0 +1,304 @@
+	log.Errorf(h.errPrefix+": debug client error: %s", err)
+}
+func (h *mockHandler) GlobalAnnouncement(message string) {
+	log.Debugf("GLOBAL ANNOUNCEMENT: %s", message)
+}
+
+func TestVuvuzela(t *testing.T) {
+	u := createVuvuzelaUniverse()
+	defer func() {
+		// Give time for goroutines to finish before pulling the rug out from under them
+		time.Sleep(1 * time.Second)
+		u.Destroy()
+	}()
+
+	aliceClient := u.createUser("alice")
+	_, err := aliceClient.ConnectConvo()
+	if err != nil {
+		t.Fatalf("Failed to connect Alice's conversation: %s", err)
+	}
+	defer aliceClient.CloseConvo()
+
+	bobClient := u.createUser("bob")
+	_, err = bobClient.ConnectConvo()
+	if err != nil {
+		t.Fatalf("Failed to connect Bob's conversation: %s", err)
+	}
+	defer bobClient.CloseConvo()
+
+	aliceRoundAndDeadDropMsgs := <-aliceClient.Handler.(*mockHandler).sentOutgoing
+	aliceRound := aliceRoundAndDeadDropMsgs.round
+	aliceMsgs := aliceRoundAndDeadDropMsgs.messages
+
+	bobRoundAndDeadDropMsgs := <-bobClient.Handler.(*mockHandler).sentOutgoing
+	bobRound := bobRoundAndDeadDropMsgs.round
+	bobMsgs := bobRoundAndDeadDropMsgs.messages
+
+	aliceRoundAndReplies := <-aliceClient.Handler.(*mockHandler).sentReplies
+	aliceReplyRound := aliceRoundAndReplies.round
+	aliceReplies := aliceRoundAndReplies.replies
+
+	bobRoundAndReplies := <-bobClient.Handler.(*mockHandler).sentReplies
+	bobReplyRound := bobRoundAndReplies.round
+	bobReplies := bobRoundAndReplies.replies
+
+	if len(bobMsgs) != 1 {
+		t.Fatalf("bob should have only sent 1 message")
+	}
+	if len(bobMsgs) != len(aliceReplies) {
+		t.Fatalf("bob should have as many messages as alice sent")
+	}
+	if len(aliceMsgs) != 1 {
+		t.Fatalf("alice should have only sent 1 message")
+	}
+	if len(aliceMsgs) != len(bobReplies) {
+		t.Fatalf("alice should have as many messages as bob sent")
+	}
+
+	latestRound, err := aliceClient.LatestRound()
+	if err != nil {
+		t.Fatalf("Failed to get latest round: %s", err)
+	}
+	if bobRound > latestRound || aliceRound > latestRound || aliceReplyRound > latestRound || bobReplyRound > latestRound {
+		t.Fatalf("Latest round %d should be same or later than %d, %d %d, %d", latestRound, bobRound, aliceRound, aliceReplyRound, bobReplyRound)
+	}
+
+	prevConvoConfig := u.ConvoConfig
+	prevConvoConfigInner := prevConvoConfig.Inner.(*convo.ConvoConfig)
+	newConvoConfig := &config.SignedConfig{
+		Version:        config.SignedConfigVersion,
+		Created:        time.Now(),
+		Expires:        time.Now().Add(24 * time.Hour),
+		PrevConfigHash: prevConvoConfig.Hash(),
+		Guardians: []config.Guardian{
+			{
+				Username: "claire",
+				Key:      u.GuardianPub,
+			},
+		},
+		Service: "Convo",
+		Inner: &convo.ConvoConfig{
+			Version: convo.ConvoConfigVersion,
+			Coordinator: convo.CoordinatorConfig{
+				Key:     prevConvoConfigInner.Coordinator.Key,
+				Address: prevConvoConfigInner.Coordinator.Address,
+			},
+			MixServers: mock.LaunchMixchain(2, u.CoordinatorPub).Servers,
```

Comment:
Creating a new mixchain should be on its own line, separate from the config.

Reply:
makes sense, amended